pkg/upgrade/upgrades/tenant_table_migration.go

```diff
@@ -80,6 +71,37 @@ SELECT n2.id
 	}
 
 	_, err = d.InternalExecutor.ExecEx(ctx, "add-system-entry", nil,
-		sessiondata.NodeUserSessionDataOverride, addSystemTenantEntry)
+		sessiondata.NodeUserSessionDataOverride,
+		`UPSERT INTO system.public.tenants (id, active, info, name, data_state, service_mode)
+VALUES (1, true,
+	crdb_internal.json_to_pb('cockroach.multitenant.TenantInfo', '{"deprecatedId":1,"deprecatedDataState":"READY"}'),
+  '`+catconstants.SystemTenantName+`',
+  `+strconv.Itoa(int(mtinfopb.DataStateReady))+`,
+  `+strconv.Itoa(int(mtinfopb.ServiceModeShared))+`)`,
```

Comment:
The assumption baked into this migration is that no other tenants have names. Is that legit? Maybe comment that somewhere